lib/core/include/irods/base64.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef IRODS_BASE64_HPP
+#define IRODS_BASE64_HPP
+
+/// \file
+
+namespace irods
+{
+    /// Base64 encode a buffer (NUL terminated)
+    ///
+    /// \param[in]      The input buffer to encode
+    /// \param[in]      The length of the input buffer
+    /// \param[out]     The destination of the base64 encoded data
+    /// \param[in,out]  The max size and resulting size
+    ///
+    /// \return 0 if successful
+    ///
+    /// \since 4.3.1
+    int base64_encode(const unsigned char* in, unsigned long inlen, unsigned char* out, unsigned long* outlen);
+
+    /// Base64 decode a block of memory
+    /// \param[in]      The base64 data to decode
```

Comment:
Let's add a newline between these lines so that doxygen doesn't get tripped up by it and so that it matches the documentation on the function above.